core/src/main/java/org/infinispan/transaction/impl/WriteSkewHelper.java

```diff
@@ -57,52 +61,62 @@ public static EntryVersionsMap performWriteSkewCheckAndReturnNewVersions(Version
                }
                VersionedRepeatableReadEntry entry = (VersionedRepeatableReadEntry) cacheEntry;
 
-               if (entry.performWriteSkewCheck(dataContainer, segment, persistenceManager, context,
-                                               prepareCommand.getVersionsSeen().get(k), versionGenerator, timeService)) {
-                  IncrementableEntryVersion oldVersion = (IncrementableEntryVersion) entry.getMetadata().version();
-                  IncrementableEntryVersion newVersion = entry.isCreated() || oldVersion == null
-                        ? versionGenerator.generateNew()
-                        : versionGenerator.increment(oldVersion);
-                  uv.put(k, newVersion);
-               } else {
-                  // Write skew check detected!
-                  throw new WriteSkewException("Write skew detected on key " + k + " for transaction " +
-                                                     context.getCacheTransaction(), k);
-               }
+               CompletionStage<Boolean> skewStage = entry.performWriteSkewCheck(entryLoader, segment, context,
```

Comment:
Nothing is inserted into the ctx by the underlying method. They are only read. I am assuming you are more coming from the aspect of future code changes could assume they can write to the context. This would require some additional changes I wasn't trying to mess with. But I would say a separate JIRA might be better.